### PR TITLE
CORE-5615 - Add metrics for link manager

### PR DESCRIPTION
**Pull request by dimosr**
## Changes
Adding some basic metrics for the link manager processor.

## Testing
Tested by performing a deployment, doing registration of a few members and checking the metrics on Grafana.
Screenshot 2023-02-20 at 15 53 06

### CODE REVIEW

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/SessionManagerImpl.kt [hunk 1]
```diff
@@ -1084,5 +1086,13 @@ internal class SessionManagerImpl(
         private fun timeStamp(): Long {
             return clock.instant().toEpochMilli()
         }
+
+        private fun recordSessionTimeoutMetric(source: HoldingIdentity, destination: HoldingIdentity) {
```
Comment (williamvigorr3) [inline components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/SessionManagerImpl.kt:1099]: We could also track the total number of outbound/inbound sessions.

**Review by williamvigorr3 — DISMISSED** [review]
I left a comment.

File: components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/SessionManagerImpl.kt (hunk 1 above)
Comment (dimosr) [reply components/link-manager/src/main/kotlin/net/corda/p2p/linkmanager/sessions/SessionManagerImpl.kt:1099]: I did this. I gave a quick try to segment with tags by source/destination, but with the current data structures in place it's a bit tricky (and also for inbound we don't necessarily know the source yet). So, I only added a metric for the total number for now.

**Review by williamvigorr3 — APPROVED** [review]
LGTM

**Pull request merged**